src/dotnet/Program.cs

```diff
@@ -36,6 +37,7 @@ public class Program
             ["build"] = BuildCommand.Run,
             ["clean"] = CleanCommand.Run,
             ["help"] = HelpCommand.Run,
+            ["list"] = (new ListCommand()).Run,
```

Comment:
`dotnet add`, `dotnet remove`, `dotnet list` are similar to each other but not really similar to anything else but `dotnet` - they represent a group of command and do share a lot of code while other commands are mostly different to each other. They are different to `dotnet * p2p` commands which also are not similar to each other and they do follow the general pattern.

In cases where Run invokation does depend on some shared properties it makes sense to have it be non-static in others it doesn't - I don't see how this is not being inconsistent.

Following this pattern of "inconsistencies" you could say the same way that we should or should not ever use `static` keyword because it is being inconsistent with other places where we don't use it - we can add hundreds of lines of extra logic for the sake of having consistency in these 10 lines but IMO it does not make any sense - these commands are different and thus are different in the code. Does it make sense to convert all static methods to non-static or reverse for the sake of consistency?